### ISSUE TITLE
[data grid] `onRowDoubleClick` not working on custom rendered cells

### ISSUE THREAD

**Issue by AgrYpn1a** [mui/mui-x#7169]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

Steps:
1. Define custom `cellRender` for columns
2. Try to trigger `onRowDoubleClick`  - it won't work

### Current behavior 😯

The`onRowDoubleClick` is not firing when double click on a cell from the row.

### Expected behavior 🤔

The`onRowDoubleClick` should fire when double click on the cell from the row, like it does on non-custom rendered cells.

### Context 🔦

I have created a hoc to wrap my cell content with so that I can show tooltip when necessary

```ts
export const cellWithTooltip = <T extends GridRenderCellParams>(RenderCell: React.FC<T>) => {
    // eslint-disable-next-line react/display-name
    return (props: T & CellWithTooltipProps) => {
        const cellContentRef = useRef<HTMLDivElement>(null);
        const [cellContentWidth, setCellContentWidth] = useState(0);

        useEffect(() => {
            cellContentRef.current && setCellContentWidth(cellContentRef.current.clientWidth);
        }, [cellContentRef.current?.clientWidth]);

        const cellWidth = props.colDef.computedWidth!;
        const enableHover = props.alwaysShowTooltip || cellContentWidth + (props.cellBoundsTolerance || 0) > cellWidth;

        return (
            <LightTooltip
                disableHoverListener={!enableHover}
                title={props.tooltipTitle}
                style={{
                    width: '100%',
                }}>
                <RenderCell ref={cellContentRef} {...props} />
            </LightTooltip>
        );
    };
};
```

And then I am using it to configure column rendering

```ts
export const getBasicFieldConfig = (field: SectionFieldMetaData, parentField?: SectionFieldMetaData): DataGridColDef => ({
    field: field.columnName,
    headerName: getFieldLabel(field, parentField),
    filterType: field.filterType as FilterType,
    minWidth: MIN_TABLE_COLUMN_WIDTH,
    flex: 1,
    valueGetter: (params: GridValueGetterParams) => {
        const value = params.value && (params.value as EntryField).simpleValue;
        return value && value.join(', ');
    },
    renderCell: (params) => {
        const value = params.formattedValue || params.value || '';
        const RenderColumn = cellWithPlainTextTooltip(
            React.forwardRef<HTMLDivElement, GridRenderCellParams>((_params, ref) => {
                return <CellContentPlainText ref={ref}>{value}</CellContentPlainText>;
            }),
        );

        return <RenderColumn {...params} tooltipTitle={<div dangerouslySetInnerHTML={{__html: value}} />} />;
    },
});

```

and wherever I define cells in similar way, the click events stop working. 

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by AgrYpn1a** [comment]
@m4theushw Hey. thanks for sharing the template, I just filled it up and as you'll be able to see double click will work on the Column 2 but not on Column 1 which has custom cell render

**Comment by AgrYpn1a** [comment]
@m4theushw I meant, on my side, on my local machine after making sure the components are not recreated each time

**Comment by AgrYpn1a** [comment]
@m4theushw Hey I figured it out, there were still too many re-renders so I further optimised my render components with `React.memo` and `React.useCallback` and got it working, closing this now.

Note for anyone that may experience similar issues, use the React Dev Tools, Profiler to catch the re-renders and see the reasons. The `colDef` field usually changed and caused re-renders so I had to exclude it from passing it as a prop and instead I only selected individual things that I really needed.
